Assignment 2.Rmd

```diff
@@ -75,20 +98,163 @@ plot(g,layout=layout.fruchterman.reingold)
+                 'DistributionChannel','UserLanguage','Q9','Q10','Q16','Q17') #remove unnecessary columns
+
+classdata <- classdata[ , !(names(classdata) %in% dropcolumns)] #remove unnecessary columns
+
+classdata <- setnames(classdata, old=c('Q8','Q1','Q3','Q4','Q5','Q6','Q7'), new=c("StudentName","Class1",'Class2','Class3','Class4','Class5','Class6')) #rename column names
+
+sapply(classdata, class) #check to see classtype in columns
+
+# making all classes uniform
+
+classdata$Class1 <- sub(" ", "", classdata$Class1) #make Class1 classes uniform
+classdata$Class1 
+classdata[34, 2] = 'HUDM4050' #edit class name zimo
+classdata$Class1
+
+classdata$Class2 <- sub(" ", "", classdata$Class2) #make Class2 uniform
+classdata$Class2
+classdata[14,3] = "IFSF4090" #edit class name
+classdata[34,3] = "HUDM4125"
+classdata$Class2
+
+classdata$Class3 <- sub(" ", "", classdata$Class3) #make Class3 uniform
+classdata$Class3
+classdata[14,4] = 'EDPS4002' #edit class name
+classdata[18,4] = 'QMSS5067'
+classdata[21,4] = NA
+classdata[34,4] = 'HUDM5026'
+classdata[45,4] = 'QMSS5072'
+classdata$Class3
+
+classdata$Class4 <- sub(" ", "", classdata$Class4) #make Class4 uniform
+classdata$Class4
+classdata[5,5] = 'QMSS5067'
+classdata[14,5] = 'EDPS4021'
+classdata[18,5] = 'QMSS5072'
+classdata[34,5] = 'HUDM5126'
+classdata$Class4
+
+classdata$Class5 <- sub(" ", "", classdata$Class5) #make Class4 uniform
+classdata$Class5
+classdata[22,6] = 'QMSS5015'
+classdata$Class5
+
+classdata$Class6 <- sub(" ", "", classdata$Class6) #make Class4 uniform
+classdata$Class6
+
+classdata[is.na(classdata)] <- 0
+
+classdata1 <- classdata %>% 
+  pivot_longer(cols = -StudentName, values_drop_na = TRUE) %>% 
+  distinct(StudentName, value) %>% 
+  mutate(n = 1) %>% 
+  pivot_wider(names_from = value, values_from= n)
+
+classdata1[is.na(classdata1)] <- 0
+classdata1 <- classdata1[,2:55]
+classdata1 <- as.matrix(classdata1)
+classdata1 
+
+rownames(classdata) <- classdata$StudentName
+dim(classdata1)
+dim(t(classdata1))
+ 
+stubystumatrix <- classdata1 %*% t(classdata1)    
+
+rownames(stubystumatrix) <- rownames(classdata)
+colnames(stubystumatrix) <- rownames(classdata)
+stubystumatrix
+
+classmatrix <- graph_from_adjacency_matrix(stubystumatrix, mode = c("undirected"), weighted = NULL,
+                            diag = FALSE, add.colnames = NULL, add.rownames = NA)
+
+plot(classmatrix)
+
+ # classdata1[is.na(classdata1)] <- 0
+ # classdata1 <- classdata1[,2:55]
+ # classdata1 <- data.matrix(classdata1)
+
+  
+#dim(classdata1)
+#dim(t(classdata1))
+
+#classdata1 %*% t(classdata1)    
+
+
+
+#classdata2 <- classdata %>% 
+#  pivot_longer(cols = -StudentName, values_drop_na = TRUE) %>% 
+#  distinct(StudentName, value) 
+#classdata2 
+
+
+```
+
+
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
```

Comment:
Overall, great job! Try to find the centrality measures for this network and interpret them within the context of this dataset.